patches/chrome-browser-android-tab_features_android.h.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/chrome/browser/android/tab_features_android.h b/chrome/browser/android/tab_features_android.h
+index 03c4880f2f1b0..d3c9a5b87ecf4 100644
+--- a/chrome/browser/android/tab_features_android.h
++++ b/chrome/browser/android/tab_features_android.h
+@@ -22,7 +22,7 @@ class SyncSessionsRouterTabHelper;
+ class TabFeaturesAndroid {
+  public:
+   TabFeaturesAndroid(content::WebContents* web_contents, Profile* profile);
+-  ~TabFeaturesAndroid();
++  virtual ~TabFeaturesAndroid();
```

Comment:
Maybe we should rename the base class so we don't need this patch. So change TabFeaturesAndroid -> TabFeaturesAndroid_ChromiumImpl and call our subclass TabFeaturesAndroid so there are no base class pointers? If we do that here we should do the same for TabFeatures for consistency

Reply:
actually we can still name ours `BraveTabFeatures*` and typedef/using to also be `TabFeatures*`

Reply:
I think its better to subclass here - that way if upstream adds some new public methods (highly likely, judging by the look of `TabFeatures` on desktop) it won't break our overrides.

Additionally if we do that we'll need a `std::unique_ptr<TabFeaturesAndroid_ChromiumImpl>` on our class to ensure everything its instantiating still exists?

Reply:
we do this in other places as well when having a subclass with a different name is problematic for some reason

Reply:
We'll still need to make the destructor virtual though, right? Or would it be safe because all the upstream code would be dealing with our instance of the class?

Reply:
Hey, so I had a quick go at converting this over - we can't declare ours to be `BraveTabFeatures` and have a using statement for `TabFeatures` because it conflicts with the forward declarations in `TabAndroid`.

I'll test having ours as `TabFeaturesAndroid` but personally I think I'd prefer having the patch - especially because I suspect that they'll convert this constructor to virtual once they flesh out the functionality of `TabFeaturesAndroid` a bit more.

I suspect that class is going to see a bit of churn because:
1. It doesn't support being replaced in tests
2. It has extremely limited functionality (when compared to the desktop equivalent)
3. It doesn't seem like it can replace TabHelpers yet

Reply:
I've pushed the replacement - I wasn't sure whether to leave the filenames as `brave_` or not so I've left it for now.

FWIW, I still have a preference for the patching approach for the reasons above